internal/codec/pomelo_packet_decoder_test.go

```diff
@@ -0,0 +1,82 @@
+package codec
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/topfreegames/pitaya/constants"
+	"github.com/topfreegames/pitaya/internal/packet"
+)
+
+var forwardTables = map[string]struct {
+	buf []byte
+	err error
+}{
+	"test_ok":                   {[]byte{packet.Handshake}, nil},
+	"test_wrong_packet_type":    {[]byte{0x06}, packet.ErrWrongPomeloPacketType},
+	"test_over_max_packet_size": {[]byte{packet.Handshake, 0xFF, 0xFF, 0xFF}, ErrPacketSizeExcced},
+}
+
+var (
+	// headers
+	handshakeHeader = []byte{packet.Handshake, 0x00, 0x00, 0x01}
+	invalidHeader   = []byte{0x06, 0x00, 0x00, 0x01}
+)
+
+var decodeTables = map[string][]struct {
+	data   []byte
+	packet []*packet.Packet
+	err    error
+}{
+	"test_not_enough_bytes":       {{[]byte{0x01}, nil, constants.ErrMessageWithNotEnoughLength}},
```

Comment:
it would be nice (and also seems easy) to create a case that would throw ErrPacketSizeExcced

Reply:
also, create cases that cover all packet.Type on https://github.com/topfreegames/pitaya/blob/master/internal/packet/constants.go

Reply:
I didn't created a test for ErrPacketSizeExcced because it is already tested on TestForward. So on TestDecode, it assumes forward is correct (and tested) and just returns some error to fall on this error case on Decode.

Reply:
Ok, makes sense to create a test case for each packet.Type